source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -285,7 +285,8 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
   const bool isCompressible =
       isEnabledAndContentLengthBigEnough && !Http::Utility::isUpgrade(headers) &&
       config.isContentTypeAllowed(headers) && !hasCacheControlNoTransform(headers) &&
-      isEtagAllowed(headers) && !headers.getInline(response_content_encoding_handle.handle());
+      isEtagAllowed(headers) && !headers.getInline(response_content_encoding_handle.handle()) &&
+      headers.get(Http::Headers::get().ContentRange).empty();
```

Comment:
Instead of checking for a `content-range` header, we should just refuse to compress 206 responses. Compressing a 206 response produces a response that has no defined meaning. Unless we know intimate details of the specific origin and/or client, there's no viable way to compress a 206 response.

Reply:
thanks for advise. i check the 206 status spec, it says:
> To avoid confusion with single-part responses, a server MUST NOT generate a [Content-Range](https://www.rfc-editor.org/rfc/rfc9110#field.content-range) header field in the HTTP header section of a multiple part response (this field will be sent in each part instead).
so you are right, we should check status code 206 instead of the content-range header.

Reply:
done